Content.Shared/Movement/Systems/SharedJetpackSystem.cs

```diff
@@ -0,0 +1,137 @@
+using Content.Shared.Actions;
+using Content.Shared.Hands;
+using Content.Shared.Inventory;
+using Content.Shared.Inventory.Events;
+using Content.Shared.Movement.Components;
+using Content.Shared.Movement.Events;
+using Content.Shared.Toggleable;
+using Robust.Shared.Containers;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.Movement.Systems;
+
+public abstract class SharedJetpackSystem : EntitySystem
+{
+    [Dependency] protected readonly SharedContainerSystem Container = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<JetpackComponent, GetItemActionsEvent>(OnJetpackGetAction);
+        SubscribeLocalEvent<JetpackComponent, ToggleJetpackEvent>(OnJetpackToggle);
+        SubscribeLocalEvent<JetpackComponent, GotEquippedEvent>(OnJetpackEquipped);
+        SubscribeLocalEvent<JetpackComponent, GotEquippedHandEvent>(OnJetpackHandEquipped);
+        SubscribeLocalEvent<JetpackComponent, GotUnequippedEvent>(OnJetpackUnequipped);
+        SubscribeLocalEvent<JetpackComponent, GotUnequippedHandEvent>(OnJetpackHandUnequipped);
+
+        SubscribeLocalEvent<JetpackUserComponent, CanWeightlessMoveEvent>(OnJetpackUserCanWeightless);
+        SubscribeLocalEvent<JetpackUserComponent, MobMovementProfileEvent>(OnJetpackUserMovement);
+    }
+
+    private void OnJetpackUserMovement(EntityUid uid, JetpackUserComponent component, ref MobMovementProfileEvent args)
+    {
+        // Only overwrite jetpack movement if they're offgrid.
+        if (args.Override || !args.Weightless) return;
+
+        args.Override = true;
+        args.Acceleration = component.Acceleration;
+        args.WeightlessModifier = component.WeightlessModifier;
+        args.Friction = component.Friction;
+    }
+
+    private void OnJetpackUserCanWeightless(EntityUid uid, JetpackUserComponent component, ref CanWeightlessMoveEvent args)
+    {
+        args.CanMove = true;
+    }
+
+    private void OnJetpackHandUnequipped(EntityUid uid, JetpackComponent component, GotUnequippedHandEvent args)
+    {
+        if (!component.Enabled) return;
+        RemComp<JetpackUserComponent>(args.User);
+    }
+
+    private void OnJetpackUnequipped(EntityUid uid, JetpackComponent component, GotUnequippedEvent args)
+    {
+        if (!component.Enabled) return;
+        RemComp<JetpackUserComponent>(args.Equipee);
+    }
```

Comment:
shouldnt jetpackuser bookkeeping be handled by the action enabling/disabling. cause its an item action, it already does this stuff

Reply:
same with the handequip/equipped stuff below

Reply:
Is there even an event for it? Internals just subscribes to inventory events.